projects/packages/waf/src/class-waf-runtime.php

```diff
@@ -247,6 +247,73 @@ public function block( $action, $rule_id, $reason, $status_code = 403 ) {
 		}
 	}
 
+	/**
+	 * Write block logs.
+	 *
+	 * @param string $rule_id Rule id.
+	 * @param string $reason Block reason.
+	 */
+	public function write_blocklog( $rule_id, $reason ) {
+		$log_data              = array();
+		$log_data['rule_id']   = $rule_id;
+		$log_data['reason']    = $reason;
+		$log_data['timestamp'] = gmdate( 'Y-m-d H:i:s' );
+
+		$file_path = JETPACK_WAF_DIR . '/waf-blocklog';
+
+		$fp = fopen( $file_path, 'a+' );
+		try {
+			fwrite( $fp, json_encode( $log_data ) . "\n" );
```

Comment:
The Scan side will delete this file once ingested, right? I still wonder if we should add a filesize check here just so we don't kill their FS space if the remote scan isn't working.

Reply:
Yep, but a check seems like a reasonable safety step. I'll add it. Any suggestions on a max size?

Reply:
Not sure; I'd see if you get a sense of what the average per message is over, say 100 block log lines, and then set it at a level where we hopefully won't lose log data daily on busy sites, but won't use up all the space on a shared host site. 1GB? 500MB?

Reply:
I added a check. 100 mb to start with, we can always adjust later if we think it's not enough.